asdf/asdf.py

```diff
@@ -1287,6 +1222,132 @@ def get_history_entries(self):
+        file-like object.
+
+    uri : string, optional
+        The URI of the file.  Only required if the URI can not be
+        automatically determined from `fd`.
+
+    mode : string, optional
+        The mode to open the file in.  Must be ``r`` (default) or
+        ``rw``.
+
+    validate_checksums : bool, optional
+        If `True`, validate the blocks against their checksums.
+        Requires reading the entire file, so disabled by default.
+
+    extensions : list of AsdfExtension
+        A list of extensions to use when reading and writing ASDF files.
+        See `~asdf.asdftypes.AsdfExtension` for more information.
+
+    do_not_fill_defaults : bool, optional
+        When `True`, do not fill in missing default values.
+
+    ignore_version_mismatch : bool, optional
+        When `True`, do not raise warnings for mismatched schema versions.
+        Set to `True` by default.
+
+    ignore_unrecognized_tag : bool, optional
+        When `True`, do not raise warnings for unrecognized tags. Set to
+        `False` by default.
+
+    copy_arrays : bool, optional
+        When `False`, when reading files, attempt to memmap underlying data
+        arrays when possible.
+
+    lazy_load : bool, optional
+        When `True` and the underlying file handle is seekable, data
+        arrays will only be loaded lazily: i.e. when they are accessed
+        for the first time. In this case the underlying file must stay
+        open during the lifetime of the tree. Setting to False causes
+        all data arrays to be loaded up front, which means that they
+        can be accessed even after the underlying file is closed.
+        Note: even if `lazy_load` is `False`, `copy_arrays` is still taken
+        into account.
+
+    custom_schema : str, optional
+        Path to a custom schema file that will be used for a secondary
+        validation pass. This can be used to ensure that particular ASDF
+        files follow custom conventions beyond those enforced by the
+        standard.
+
+    strict_extension_check : bool, optional
+        When `True`, if the given ASDF file contains metadata about the
+        extensions used to create it, and if those extensions are not
+        installed, opening the file will fail. When `False`, opening a file
+        under such conditions will cause only a warning. Defaults to
+        `False`.
+
+    ignore_missing_extensions : bool, optional
+        When `True`, do not raise warnings when a file is read that
+        contains metadata about extensions that are not available. Defaults
+        to `False`.
+
+    Returns
+    -------
+    asdffile : AsdfFile
+        The new AsdfFile object.
+    """
+
+    # For now retain backwards compatibility with the old API behavior,
+    # specifically when being called from AsdfFile.open
+    if not _compat:
+        mode = _check_and_set_mode(fd, mode)
+        if mode == 'r' and not copy_arrays:
+            copy_arrays = True
```

Comment:
Should we print a warning here?

Reply:
Yes, but I just pushed a new set of changes that makes this irrelevant. I think the update contains a better solution to the issue you reported.